Client/Homepage/script.js

```diff
@@ -0,0 +1,27 @@
+// SETUP
+const searchBtn = document.getElementById("searchBtn");
+const luckyBtn = document.getElementById("luckyBtn");
+
+//*******************************************************
+//Search button flow
+const searchFunc = async() => {
+    //click and show search results page
+
+  // const url = await response.text();
+   window.location.replace('../Search Page/index.html')
+
+
+  //window.location.href = response;
+};
+
+//*******************************************************
+//Random button flow
+
+const luckyFunc = async() => {
+    const response = await fetch('http://localhost:3000/random'); //fetches promise from server at random endpoint
+    const url = await response.text(); //translates promise to text using .text() method and assign string to url variable
+    window.location.href = url; //new window set to url
+};
+
+searchBtn.addEventListener("click", searchFunc);
+luckyBtn.addEventListener("click", luckyFunc);
```

Comment:
well commented
since lucky button does not appear to work, could try onload="luckyFunc()" in the html
one way to pass data to use the api fetch in the search page is using sessionStorage